app/grants/views.py

```diff
@@ -922,26 +923,47 @@ def grants_cart_view(request):
 
 
 def grants_bulk_add(request, grant_str):
-    
+    grants = {}
     redis = RedisService().redis
     key = hashlib.md5(grant_str.encode('utf')).hexdigest()
     views = redis.incr(key)
 
-    grant_ids = grant_str.split(':')[0].split(',')
-    grant_ids = [int(ele) for ele in grant_ids if ele and ele.isnumeric() ]
+    grants_data = grant_str.split(':')[0].split(',')
+
+    for ele in grants_data:
+        # new format will support amount and token in the URL separated by ;
+        grant_data = ele.split(';')
```

Comment:
is this backwards compatible with the old links that have already been shared out in the wild?